File: lecture1/primes.py
from sys import argv
from math import sqrt

limit = int(argv[1])
count = 0

for n in range(2, limit + 1):
    prime = True
    root = int(sqrt(n))
    
    for d in range(2,n):
        if n % d == 0:
            prime = False
            break
        if prime:
            count += 1

print(count)            

File: lecture1/primes.cpp
#include <iostream>
#include <cstdlib>
#include <cmath>

using namespace std;

int main(int argc, char* argv[]){
    int limit = atoi[argv[1]];
    int count = 0;

    for(int n = 2; n <= limit; n++){
        bool prime = true;
        int root = (int) sqrt(n);

        for(int d = 2; d <= root; d++){
            if( n % d == 0){
                prime = false;
                break;
            }
        }
        if(prime){
            count++;
        }
    }
    cout << count << endl;
    return 0;
}